Assets/Features/Building/Scripts/Grid/Cursor.cs

```diff
@@ -0,0 +1,198 @@
+        public void Release(Vector3Int position)
+        {
+            _currentBrush.Release(position);
+        }
+        
+        /// <summary>
+        /// Called when not releasing or holding to show the current location
+        /// </summary>
+        /// <param name="position">Position of the hover</param>
+        public void Hover(Vector3Int position)
+        {
+            _currentBrush.Hover(position);
+        }
+        
+        /// <summary>
+        /// Rotates the selected building
+        /// </summary>
+        /// <param name="direction"></param>
+        public void Rotate(int direction)
+        {
+            _currentBrush.Rotate(direction);
+        }
+        
+        /// <summary>
+        /// Called when the cursor should be visualized
+        /// </summary>
+        public void UpdateVisuals()
+        {
+            Visualize(_currentBrush.SelectedTiles, _currentBrush.RequireAll);
+        }
+
+        private void Update()
+        {
+            //enable cursor when over UI
+            _cursorTilemap.gameObject.SetActive(!_eventSystem.IsPointerOverGameObject());
+        }
+        
+        /// <summary>
+        /// Changes a world position to a clamped grid position
+        /// </summary>
+        /// <param name="worldPosition">Position to be clamped</param>
+        /// <returns></returns>
+        public Vector3Int WorldToGirdPosition(Vector3 worldPosition)
+        {
+            Vector3 pos = worldPosition;
+
+            //clamp to grid positions
+            Vector2 clampedValue = new Vector2(RoundToMultiple(pos.x, _grid.CellSize),
+                RoundToMultiple(pos.y, _grid.CellSize));
+
+            return _grid.ClampedWorldToGridPosition(clampedValue,
+                (int)_currentSelectedBuilding.BuildItems[0].GridPosition.Layer);
+        }
+
+        /// <summary>
+        /// Updates the cursor tilemap, so it visualizes the current selected tiles
+        /// </summary>
+        /// <param name="selectedGroup">The current selected tiles</param>
+        /// <param name="requireAllAvailable">Sets that all need to be required if placed</param>
+        /// <param name="flipColors">Flip colors to the correct and incorrect</param>
+        private void Visualize(List<SubBuildItem> selectedGroup, bool requireAllAvailable = false,
+            bool flipColors = false)
+        {
+            //sets the offset of the whole grid
+            Vector3Int offset = new Vector3Int(Mathf.RoundToInt(_cursorTilemap.transform.position.x),
+                Mathf.RoundToInt(_cursorTilemap.transform.position.y),
+                Mathf.RoundToInt(_cursorTilemap.transform.position.z));
+
+            _cursorTilemap.ClearAllTiles();
+
+            //check if all positions are required when placing
+            bool valid = true;
+            if (requireAllAvailable)
+            {
+                foreach (var gridPosition in selectedGroup)
```

Comment:
Rather use Vec2 / Vec3 here instead of `var` since I don't know which of the two this is.